dialer.go

```diff
@@ -12,6 +13,10 @@ import (
 	"github.com/streadway/amqp"
 )
 
+const (
+	errChanIsUnbuffered = "receiver chan is unbuffered"
```

Comment:
create separate errors:

```go
const (
  errReadyChanUnbuffered = fmt.Errorf("ready chan is unbuffered")
  errUnreadyChanUnbuffered = fmt.Errorf("unready chan is unbuffered")
)
```

Reply:
fmt.Errorf() returns an error then cannot be used as const. I suggest use something like errMsgDescription for this approach

Reply:
errors.New()

Reply:
error msgs separated